bindings/python/multibody/geometry-object.hpp

```diff
@@ -28,27 +28,44 @@ namespace pinocchio
       {
         cl
         .def(bp::init<std::string,FrameIndex,JointIndex,CollisionGeometryPtr,SE3,
-                      bp::optional<std::string,Eigen::Vector3d,bool,Eigen::Vector4d,std::string> >(
-             bp::args("name","parent frame index","parent joint index","FCL collision geometry",
-                      "placement", "meshPath", "meshScale", "overrideMaterial", "meshColor", "meshTexturePath")))
+                      bp::optional<std::string,Eigen::Vector3d,bool,Eigen::Vector4d,std::string> >
+             (
+             bp::args("self","name","parent_frame index","parent_joint index","collision_geometry",
+                      "placement", "meshPath", "meshScale", "overrideMaterial", "meshColor", "meshTexturePath"),
+             "Full constructor of a GeometryObject."))
+        .def(bp::init<std::string,JointIndex,CollisionGeometryPtr,SE3,
+                      bp::optional<std::string,Eigen::Vector3d,bool,Eigen::Vector4d,std::string> >
+             (
+              bp::args("self","name","parent_joint index","collision_geometry",
+                      "placement", "meshPath", "meshScale", "overrideMaterial", "meshColor", "meshTexturePath"),
+              "Reduced constructor of a GeometryObject. This constructor does not require to specify the parent frame index."
+              ))
         .add_property("meshScale",
                       bp::make_getter(&GeometryObject::meshScale,
                                       bp::return_value_policy<bp::return_by_value>()),
                       bp::make_setter(&GeometryObject::meshScale),
-                      "Scaling parameter for the mesh")
+                      "Scaling parameter of the mesh.")
         .add_property("meshColor",
                       bp::make_getter(&GeometryObject::meshColor,
                                       bp::return_value_policy<bp::return_by_value>()),
                       bp::make_setter(&GeometryObject::meshColor),
-                      "Color rgba for the mesh")
-        .def_readwrite("name", &GeometryObject::name, "Name of the GeometryObject")
-        .def_readwrite("parentJoint", &GeometryObject::parentJoint, "Index of the parent joint")
-        .def_readwrite("parentFrame", &GeometryObject::parentFrame, "Index of the parent frame")
+                      "Color rgba of the mesh.")
+        .def_readwrite("fcl", &GeometryObject::fcl,
+                       "The FCL CollisionGeometry associated to the given GeometryObject.")
```

Comment:
Another unrelated note: maybe *geometry* would be a better attribute name.

Reply:
I also do not like the name of this attribute. But the name `fcl` is in C++, and I do not see how we could change it without breaking somebody's code. In Python, we should think about consistency with C++ most of all

Reply:
I think the best name for it would be geometry. @jmirabel @gabrielebndn What do you think?

Reply:
Yes. We can put this in the list of changes for Pinocchio 3.

Reply:
I think `geometry` is a good name. But again, I think it is difficult to change it in C++ without breaking code. If you are willing to do it, go on. Most of all, I would avoid having it called `fcl` in C++ and `geometry` in Python

Reply:
Or we may proceed to the change even before, because fcl is a pointer

Reply:
You mean, having two pointers to the same object, with the `fcl` pointer being deprecated?
Having two pointers, and then only using one of them, would break all code looking like `geometryObject.fcl = ...`, because it will not change `geometryObject.geometry`. I don't know how much code there may be out there doing something like this

Reply:
I will provide a solution soon ;)